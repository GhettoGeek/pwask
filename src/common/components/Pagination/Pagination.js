import React from 'react'
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core/styles'
import clsx from 'clsx'

const useStyles = makeStyles((theme) => ({
  root: {
    position: 'absolute',
    bottom: 8,
    right: 8,
    display: 'flex',
    flexDirection: 'row',
  },
  button: {
    height: 18,
    width: 18,
    cursor: 'pointer',
    border: 0,
    background: 'none',
    padding: 0,
  },
  dot: {
    backgroundColor: '#e4e6e7',
    height: 12,
    width: 12,
    borderRadius: 6,
    margin: 3,
  },
  active: {
    backgroundColor: theme.palette.primary.main,
  },
}))

function Pagination({ currentIndex, dots, onChangeIndex }) {
  const classes = useStyles()
  const handleClick = (event, index) => onChangeIndex(index)

  return dots > 1 ? (
    <div className={classes.root}>
      {[...Array(dots)].map((dot, index) => (
        <button
          type="button"
          className={classes.button}
          key={index}
          onClick={(event) => handleClick(event, index)}
        >
          <div
            className={clsx(classes.dot, {
              [classes.active]: index === currentIndex,
            })}
          />
        </button>
      ))}
    </div>
  ) : null
}

Pagination.propTypes = {
  dots: PropTypes.number,
  currentIndex: PropTypes.number.isRequired,
  onChangeIndex: PropTypes.func.isRequired,
}

Pagination.defaultProps = {
  dots: 3,
}

export default Pagination
