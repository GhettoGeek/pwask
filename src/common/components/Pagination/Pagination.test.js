import React from 'react'
import { shallow } from 'enzyme'
import Pagination from './Pagination'

describe('Pagination', () => {
  it('renders correctly the component', () => {
    const wrapper = shallow(
      <Pagination
        currentIndex={0}
        onChangeIndex={() => {}}
      />,
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
})
