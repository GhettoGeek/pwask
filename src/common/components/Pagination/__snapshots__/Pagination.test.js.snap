// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly the component 1`] = `"<div class=\\"makeStyles-root-1\\"><button type=\\"button\\" class=\\"makeStyles-button-2\\"><div class=\\"makeStyles-dot-3 makeStyles-active-4\\"></div></button><button type=\\"button\\" class=\\"makeStyles-button-2\\"><div class=\\"makeStyles-dot-3\\"></div></button><button type=\\"button\\" class=\\"makeStyles-button-2\\"><div class=\\"makeStyles-dot-3\\"></div></button></div>"`;
